File: 2000/ts/2011. Final Value of Variable After Performing Operations.ts
//1
function finalValueAfterOperations(operations: string[]): number {
    let res = 0;
    for (const opt of operations) {
        res = opt[1] === "+" ? res + 1 : res - 1;
    }
    return res;
};

//2
function finalValueAfterOperations(operations: string[]): number {
    let res=0;
    operations.forEach((el)=>{
        el.includes('+')?res++:res--;
    })
    return res
  };

File: 2000/rust/2011. Final Value of Variable After Performing Operations.rs
//1
impl Solution {
    pub fn final_value_after_operations(operations: Vec<String>) -> i32 {
        let mut res=0;
        for s in operations{
            match s.as_str() {
                "X++"|"++X"=>res+=1,
                "X--"|"--X"=>res-=1,
                _=>()
            }
        }
        res
    }
}

//2
impl Solution {
    pub fn final_value_after_operations(operations: Vec<String>) -> i32 {
        operations.iter().fold(0, 
            |res,curr| (
                if curr.as_bytes()[1]==b'+'{
                    res+1
                }else{
                    res-1
                }
            )
        )
    }
}
